src/editable-heading/editable-heading.tsx

```diff
@@ -121,15 +130,33 @@ export const EditableHeading = (props: EditableHeadingProps) => {
     setIsInSelectionMode(false);
   }, [isMouseDown, isInSelectionMode, disabled, onEdit]);
 
-  const onInputFocus = React.useCallback((e: React.FocusEvent<HTMLInputElement>) => {
-    setIsInFocus(true);
-    onFocus?.(e);
-  }, [onFocus]);
-
-  const onInputBlur = React.useCallback((e: React.FocusEvent<HTMLInputElement>) => {
-    setIsInFocus(false);
-    onBlur?.(e);
-  }, [onBlur]);
+  const onInputFocus = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+      setIsInFocus(true);
+      onFocus?.(e);
+    }, [onFocus]);
+
+  const onInputBlur = React.useCallback(
+    (e: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+      setIsInFocus(false);
+      onBlur?.(e);
+    }, [onBlur]);
+
+  textAreaRef?.current?.addEventListener('scroll', () => {
```

Comment:
I think attaching listener every render (and never removing them) isn't really a good idea, because we will end up with dozens of listeners that are trying to do the same action. Can't we put this into a useEffect?

```
  useEffect(() => {
    const onScroll = (event) => console.info("scrolling", event);
      
    window.addEventListener('scroll', onScroll);
    
    return () => {
      window.removeEventListener('scroll', onScroll);
    }
  }, []);
  ```

Reply:
Oh, yes, I am very sorry. I forgot about that, thank you

Reply:
fixed that